Sources/Controllers/2.0/OATargetInfoViewController.mm

```diff
@@ -448,6 +371,240 @@ - (BOOL) showNearestWiki
     return res;
 }
 
+- (void)sendNearbyImagesRequest:(OARowInfo *)nearbyImagesRowInfo
+{
+    OACollapsableCardsView *cardsView = (OACollapsableCardsView *)nearbyImagesRowInfo.collapsableView;
+    if (!nearbyImagesRowInfo || cardsView.cards.count > 0)
+        return;
+        
+    _cards = [NSMutableArray new];
+    [cardsView setCards:@[[[OAImageCard alloc] initWithData:@{@"key" : @"loading"}]]];
+
+    NSString *urlString = [NSString stringWithFormat:@"https://osmand.net/api/cm_place?lat=%f&lon=%f",
+                           self.location.latitude, self.location.longitude];
+    if ([self.getTargetObj isKindOfClass:OAPOI.class])
+    {
+        OAPOI *poi = self.getTargetObj;
+        NSString *imageTagContent = poi.values[@"image"];
+        NSString *mapillaryTagContent = poi.values[@"mapillary"];
+        NSString *wikimediaTagContent = poi.values[@"wikimedia_commons"];
+        NSString *wikidataTagContent = poi.values[@"wikidata"];
+        
+        if (wikimediaTagContent && ![wikimediaTagContent isEqualToString:imageTagContent])
+            [self addWikimediaImage:wikimediaTagContent poi:poi rowInfo:nearbyImagesRowInfo];
+        if (wikidataTagContent)
+            [self addWikidataImage:wikidataTagContent poi:poi rowInfo:nearbyImagesRowInfo];
+        if (imageTagContent)
+            urlString = [urlString stringByAppendingString:[NSString stringWithFormat:@"&osm_image=%@", imageTagContent]];
+        if (mapillaryTagContent)
+            urlString = [urlString stringByAppendingString:[NSString stringWithFormat:@"&osm_mapillary_key=%@", mapillaryTagContent]];
+    }
+    
+    NSURL *urlObj = [[NSURL alloc] initWithString:[[urlString stringByReplacingOccurrencesOfString:@" "  withString:@"_"] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
+    NSURLSession *aSession = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+  
+    [[aSession dataTaskWithURL:urlObj completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+        if (((NSHTTPURLResponse *)response).statusCode == 200)
+        {
+            if (data)
+            {
+                NSError *error;
+                NSString *safeCharsString = [[NSString alloc] initWithData:data encoding:NSISOLatin1StringEncoding];
+                NSData *safeCharsData = [safeCharsString dataUsingEncoding:NSUTF8StringEncoding];
+                NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:safeCharsData options:NSJSONReadingAllowFragments error:&error];
+                if (!error)
+                {
+                    for (NSDictionary *dict in jsonDict[@"features"])
```

Comment:
You should update the cards list synchronously and then sort the array/remove duplicates after all three requests are finished.